### PR TITLE
Rev to use latest go-lang and golang.org/x/net

### PR DESCRIPTION
**Pull request by robbrockbank**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -33,6 +33,88 @@ require (
 	sigs.k8s.io/kind v0.11.1 // Do not remove, not used by code but used by build
 )
 
+require (
+	cloud.google.com/go v0.81.0 // indirect
```
Comment (tmjd) [inline go.mod:37]: Woah, did something change with go 1.17 that adds all these `indirect` entries now?

Comment (robbrockbank) [reply go.mod:37]: it's shifted things around

Comment (robbrockbank) [reply go.mod:37]: and added a whole bunch more...

the second "require" section is the set of indirect imports

File: Makefile [hunk 2]
```diff
@@ -658,7 +658,9 @@ vet:
 
 # Generate code
 generate: $(BINDIR)/controller-gen
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+	$(CONTAINERIZED) $(CALICO_BUILD) sh -c '$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."'
+	# Make sure the generated code won't cause a static-checks failure.
+	$(MAKE) fix
```
Comment (doublek) [inline Makefile:663]: It's a bit surprising that this has to be run. I am okay with it, just surprised that generated code isn't formatted right.

**Review by doublek — APPROVED** [review]
LGTM. Will wait to see if @tmjd has anything to add.

File: Makefile (hunk 2 above)
Comment (robbrockbank) [reply Makefile:663]: we do this all over the place at the moment

I think the formatting adds in some extra build directives that I guess the generation code has not yet been updated to add.

File: go.mod (hunk 1 above)
Comment (robbrockbank) [reply go.mod:37]: https://stackoverflow.com/questions/69432877/why-there-are-two-require-blocks-in-go-mod-since-go-1-17

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**